internal/validate/common_password.go

```diff
@@ -0,0 +1,413 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validate
+
+var commonPasswords = map[string]bool{
```

Comment:
here I am using the same list used by the endpoint. I am not happy about this list given that we need to keep it updated with the one in the endpoint. I am happy to consider another way to validate this

Reply:
Which endpoint are you talking about? Could you please point out the endpoint?

Reply:
Instead of looking up for weak password, could we add a password policy? Something like atleast 2 uppercase, 2 lowercase, 1 number and 1 special character?

Reply:
could we make that a `private` endpoint?

Reply:
same as we did with with current IP

Reply:
> Instead of looking up for weak password, could we add a password policy?

Check for known common passwords is a good strategy, password policies are actually a bad practice now days, length and uniqueness (across services) is all that matters

Reply:
yes, I was thinking about the private endpoint but I wasn’t sure it was worth it

Reply:
@threebee this is the endpoint

`Error: POST https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/5e4e593f70dfbf1010295836/databaseUsers: 400 (request "COMMON_PASSWORD") The password provided is too weak, as it can be found in most commonly used password lists.`

Reply:
> yes, I was thinking about the private endpoint but I wasn’t sure it was worth it

I think it is but I would not block this PR on that change as we may want to discuss it further with other teams

Reply:
I am going to create a new ticket to discuss this private endpoint once I merge this